rust/core-lib/src/editor.rs

```diff
@@ -371,15 +371,11 @@ impl Editor {
                 Some(s) => s.to_owned(),
                 None => self.view.view_id.to_string(),
             };
-                let text = match new_len < MAX_SIZE_LIMIT {
-                true => Some(self.text.slice_to_string(iv.start(), iv.start() + new_len)),
-                false => None
-            };
 
```

Comment:
I'm not sure what to do about this. 😕

Reply:
Yeah, this is really really the tricky bit. It is important to keep the size of the RPC bounded, as doing something like delete all + undo on a gigabyte buffer is something that can happen. There are two ways this can go wrong; either the sizes of the strings add up to a large amount, or you can have a ton of small deltas (a million blank lines with a cursor on each, then press space).

What I'm inclined to suggest, in the interest of both correctness and simplicity, is make the entire delta optional. If it's too expensive (by a metric that correlates strongly with the size of its serialized repr), then the semantics are "invalidate everything." Probably add a non-optional field for the size of the document after delta. For plugins that compute a pure function of the input (like syntax highlighting) it's pretty easy to do the right thing. For stuff like indentation, we'll effectively drop the potential edit on the floor.

Other than that, it looks pretty good, I don't see any problems.